filter/command.go

```diff
@@ -34,18 +33,16 @@ func (o Options) Run() error {
 	v := viewport.New(o.Width, o.Height)
 
 	var choices []string
-	if input, _ := stdin.Read(); input != "" {
-		input = strings.TrimSuffix(input, "\n")
-		if input != "" {
-			choices = strings.Split(input, "\n")
+	if len(o.Options) == 0 {
+		if input, _ := stdin.Read(); input != "" {
+			o.Options = strings.Split(strings.TrimSuffix(input, "\n"), "\n")
+		} else {
+			o.Options = files.List()
 		}
-	} else {
-		choices = files.List()
 	}
 
-	if len(choices) == 0 {
-		return errors.New("no options provided, see `gum filter --help`")
-	}
+	choices = make([]string, len(o.Options))
```

Comment:
We already declared a `choices` var on line 35, any reason to re-declare here? Also why assign input values to `o.Options` instead of using `choices`? Just want to make sure I'm not missing anything :)

Reply:
Actually, when looking at it, it doesn't make sense to introduce the variable choices here. I'm changing the rest of `filter/command.go` to o.Options because o.Options is already declared and used by kong.